src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/bufferOutputPolling.ts

```diff
@@ -215,3 +240,37 @@ export async function assessOutputForErrors(buffer: string, token: CancellationT
 		return 'Error occurred ' + err;
 	}
 }
+
+/**
+ * Returns true if the last line of the output is a prompt asking for user input (e.g., "Do you want to continue? (y/n)").
+ * This is a heuristic and matches common prompt patterns.
+ */
+/**
+ * Returns the kind of input expected by the last line of the output if it appears to be a user prompt.
+ * For example, returns 'y/n', 'yes/no', 'press key', 'choice', etc., or undefined if not a prompt.
+ */
+export function getExpectedUserInputKind(output: string): string | undefined {
+	if (!output) {
+		return undefined;
+	}
+
+	const patterns: { regex: RegExp; kind: string }[] = [
+		{ regex: /\bdo you want to continue\b.*\(y\/n\)/ig, kind: 'y/n' },
```

Comment:
If we idle briefly with one of these matches, we could do more general ones like '/\(y\/n\):?\s*$/', checking the end of the line is important too.